app/commands/solution/publish.rb

```diff
@@ -2,10 +2,12 @@ class Solution
   class Publish
     include Mandate
 
-    initialize_with :solution, :iteration_idx
+    initialize_with :solution, :user_track, :iteration_idx
 
     def call
       solution.with_lock do
+        Solution::Complete.(solution, user_track) unless solution.completed?
```

Comment:
You probably need to move this outside the lock block it to work in production. Otherwise you'll try and lock it both here and in complete and I *think* that might break in production (tests don't use transactions so things like this sometimes pass in the tests and fail for real).

Reply:
Updated